servicetalk-http-netty/src/main/java/io/servicetalk/http/netty/DefaultHttpServerBuilder.java

```diff
@@ -111,20 +113,41 @@ public HttpServerBuilder bufferAllocator(final BufferAllocator allocator) {
                                              final StreamingHttpService service,
                                              final HttpExecutionStrategy strategy,
                                              final boolean drainRequestPayloadBody) {
-        final ReadOnlyHttpServerConfig roConfig = this.config.asReadOnly();
         executionContextBuilder.executionStrategy(strategy);
         final HttpExecutionContext httpExecutionContext = executionContextBuilder.build();
+
+        return doListen(connectionAcceptor, httpExecutionContext, service, drainRequestPayloadBody);
+    }
+
+    @Override
+    protected HttpExecutionContext buildExecutionContext(final HttpExecutionStrategy strategy) {
+        executionContextBuilder.executionStrategy(strategy);
+        return executionContextBuilder.build();
+    }
+
+    @Override
+    protected BooleanSupplier shouldOffload(IoExecutor ioExecutor) {
+        return ioExecutor instanceof NettyIoExecutor && ((NettyIoExecutor) ioExecutor).isIoThreadSupported() ?
```

Comment:
Instead of adding these new public methods, can `OffliadingFilter` compute the `BooleanSupplied` based on `HttpServiceContext` it receives with each request? `DefaultHttpExecutionStrategy` will need to do the same.